### ISSUE TITLE
Week number is not taken into account during date creation in `strptime`

### ISSUE THREAD

**Issue by traut** [itchyny/gojq#254]
It looks like gojq ignores the week number during `strptime` while JQ uses it for the output date:
```bash
$ jq -n '"2024, 20" | strptime("%Y, %W") | strftime("%Y-%m-%d")'
"2024-05-13"

$ gojq -n '"2024, 20" | strptime("%Y, %W") | strftime("%Y-%m-%d")'
"2024-01-01"

$ jq --version
jq-1.7

$ gojq --version
gojq 0.12.15 (rev: HEAD/go1.21.4)
```

**Comment by traut** [comment]
moreover, it seems even incorrect week numbers are not caught:
```bash
$ jq -n '"2024, 99" | strptime("%Y, %W") | strftime("%Y-%m-%d")'
jq: error (at <unknown>): date "2024, 99" does not match format "%Y, %W"

$ gojq -n '"2024, 99" | strptime("%Y, %W") | strftime("%Y-%m-%d")'
"2024-01-01"
```

**Comment by traut** [comment]
ah, [this](https://github.com/itchyny/timefmt-go/blob/8eb94fd93fc611d6ff21c384791f07db7fe417b3/parse.go#L98-L101) (from `timefmt` lib) explains the behavior -- the value for `%W` is parsed out but discarded.

**Comment by itchyny** [comment]
I have implemented parsing support for week numbers locally, will soon push to the repository.

**Comment by itchyny** [comment]
I have created a branch building with the latest timefmt-go with week number support, you can download the artifacts from [here](https://github.com/itchyny/gojq/actions/runs/9175604577).

**Comment by itchyny** [comment]
Resolved by 1324e6e320cff2b2eb9c778cfa1f960245ad41ad.
